### PR TITLE
ipatests: Added check that no Traceback found in idviews operations

### PR DESCRIPTION
**Pull request by varunmylaraiah**
https://pagure.io/freeipa/issue/7958

### CODE REVIEW

**Comment by varunmylaraiah** [comment]
@fcami Could you please review?

File: ipatests/test_integration/test_idviews.py [hunk 1]
```diff
@@ -28,6 +28,9 @@ class TestCertsInIDOverrides(IntegrationTest):
     def uninstall(cls, mh):
         super(TestCertsInIDOverrides, cls).uninstall(mh)
         cls.master.run_command(['rm', '-rf', cls.reqdir], raiseonerr=False)
+        # Checking Traceback https://pagure.io/freeipa/issue/7958
+        http_logs = cls.master.get_file_contents(paths.VAR_LOG_HTTPD_ERROR)
```
Comment (kaleemsiddiqu) [inline ipatests/test_integration/test_idviews.py]: @varunmylaraiah 
This check is just applicable to TestCertsInIDOverrides test class and not applied to TestIDViews test class.
Better would be to move this to pytest_ipa/integration/tasks.py as utility function and use that in every test class's e.g in this case TestCertsInIDOverrides and TestIDViews.

Also test out these changes with help of temp_commit.yaml for these two test classes by including these two specific tests in temp_commit.yaml.

**Comment by amore17** [comment]
For failures in test_integration/test_idviews.py::TestCertsInIDOverrides::test_certs_in_idoverrides_ad_users::setup

1. You need to define path for adcert1_file and adcert2_file same as 'cls.pwname = os.path.join(cls.reqdir, "pwd")' 
2. Also  
tasks.run_certutil(master, ['-L', '-n', cls.adcert1, '-a', '>', cls.adcert1_file], cls.reqdir)
tasks.run_certutil(master, ['-L', '-n', cls.adcert2, '-a', '>', cls.adcert2_file], cls.reqdir)
Here instead of redirecting output you can use put_file_contents()

**Comment by amore17** [comment]
LGTM you can remove temp_commit.
Also please add in commit message related pagure ticket.

File: ipatests/pytest_ipa/integration/tasks.py [hunk 2]
```diff
@@ -2196,3 +2196,12 @@ def wait_for_sssd_domain_status_online(host, timeout=120):
...
         raise RuntimeError("SSSD still offline")
+
+
+def check_traceback(host):
+    """Checking Traceback  on a remote host.
+    :param host: the host where the check takes place
+    """
+    http_logs = host.get_file_contents(paths.VAR_LOG_HTTPD_ERROR,
+                                       encoding='utf-8')
+    assert 'Traceback' not in http_logs
```
Comment (mrizwan93) [inline ipatests/pytest_ipa/integration/tasks.py:2207]: This method will not be used anywhere else than `test_idviews.py` testsuite. IMO it should be moved to `test_idviews.py` itself. 
OR 
Make it more generalize here in `tasks.py`. May be  something like:
```
def assert_string(host, string_to_check, file_name):
    """ Check the given string in specified file
    : param host: host where the check takes place
    : param string_to_check: string to be check
    : param file_name: file on which check is to be performed.
    """
    file_contents = host.get_file_contents(file_name)
    assert string_to_check in file_contents
```

Comment (varunmylaraiah) [reply ipatests/pytest_ipa/integration/tasks.py:2207]: @mrizwan93 I think we can create separate PR for refactoring work.

**Comment by varunmylaraiah** [comment]
Test run with Temp commit 
http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/fb8c738a-77ed-11ea-a54a-fa163e43b58c/

**Comment by flo-renaud** [comment]
Hi @varunmylaraiah 
I don't really see the point in adding this test. The BZ is marked as sanityOnly, meaning we don't have a reproducer.

**Comment by varunmylaraiah** [comment]
@kaleemsiddiqu I don't remember why this was asked to automate? Can you please clarify this to Flo.

**Comment by flo-renaud** [comment]
@varunmylaraiah 
A suggestion: the check is not really related to the idviews tests. It could be added at the end of all tests. In order to do that, please have a look at freeipa/ipatests/pytest_ipa/integration/__init__.py which defines test fixtures and could be enhanced.

**Comment by varunmylaraiah** [comment]
Sure Flo, will try.

**Pull request closed without merging**